utils/serde/src/main/java/io/apicurio/registry/utils/serde/JsonSchemaKafkaSerializer.java

```diff
@@ -0,0 +1,188 @@
+ * 
+ * @author eric.wittmann@gmail.com
+ */
+public class JsonSchemaKafkaSerializer<T>
+        extends AbstractKafkaStrategyAwareSerDe<SchemaValidator, JsonSchemaKafkaSerializer<T>>
+        implements Serializer<T> {
+    
+    public static final String REGISTRY_JSON_SCHEMA_SERIALIZER_VALIDATION_ENABLED = "apicurio.registry.serdes.json-schema.validation-enabled";
+
+    private static MedeiaJacksonApi api = new MedeiaJacksonApi();
+    private static ObjectMapper mapper = new ObjectMapper();
+    private static GlobalIdStrategy<SchemaValidator> latestVersionStrategy = new FindLatestIdStrategy<>();
+    
+    private boolean validationEnabled = false;
+    private SchemaCache<SchemaValidator> schemaCache;
+
+    /**
+     * Constructor.
+     */
+    public JsonSchemaKafkaSerializer() {
+    }
+
+    /**
+     * Constructor.
+     * @param client
+     */
+    public JsonSchemaKafkaSerializer(RegistryService client) {
+        super(client);
+    }
+
+    /**
+     * @see org.apache.kafka.common.serialization.Serializer#configure(java.util.Map, boolean)
+     */
+    @Override
+    public void configure(Map<String, ?> configs, boolean isKey) {
+        super.configure(configs, isKey);
+        
+        Object ve = configs.get(REGISTRY_JSON_SCHEMA_SERIALIZER_VALIDATION_ENABLED);
+        this.validationEnabled = Utils.isTrue(ve);
+        
+        this.schemaCache = new SchemaCache<SchemaValidator>(getClient()) {
+            @Override
+            protected SchemaValidator toSchema(Response response) {
+                String schema = response.readEntity(String.class);
+                return api.loadSchema(new StringSchemaSource(schema));
+            }
+        };
+    }
+    
+    /**
+     * @see org.apache.kafka.common.serialization.Serializer#serialize(java.lang.String, java.lang.Object)
+     */
+    @Override
+    public byte[] serialize(String topic, T data) {
+        // Headers are required when sending data using this serdes impl
+        throw new UnsupportedOperationException();
+    }
+    
+    /**
+     * @see org.apache.kafka.common.serialization.Serializer#serialize(java.lang.String, org.apache.kafka.common.header.Headers, java.lang.Object)
+     */
+    @Override
+    public byte[] serialize(String topic, Headers headers, T data) {
+        if (data == null) {
+            return new byte[0];
+        }
+
+        // Now serialize the data
+        try {
+            ByteArrayOutputStream baos = new ByteArrayOutputStream();
+            JsonGenerator generator = mapper.getFactory().createGenerator(baos);
+            if (validationEnabled) {
+                String artifactId = getArtifactId(topic, data);
+                Long globalId = getArtifactVersionGlobalId(artifactId, topic, data);
+                if (globalId == null) {
+                    globalId = latestVersionStrategy.findId(getClient(), artifactId, ArtifactType.JSON, null);
```

Comment:
Hmmm ... why exactly 2 tries ?

Reply:
This is weird and a holdover from the previous implementation.  I'll modify it so it makes more sense.